### ISSUE TITLE
Attribute Manipulation Generative Adversarial Networks for Fashion Images

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#168]
## TL; DR

- Localize attribute-related region with Class Activation Map (CAM) and employ additional discriminator that carefully inspects attribute-related region (path) for unpaired image-to-image translation.
- Perceptual loss is evaluated with reference image retrieved by attribute similarity.

## Motivation

- Attention-based generative models suffer when the number of attributes increase.

## Formulation

<img width="1448" alt="image" src="https://user-images.githubusercontent.com/46308407/162158477-724d25d3-2353-421c-bddd-1185a9b2899d.png">

### Generator side

- [Input image, attribute manipulation] is provided as input.
- Generator outputs coarse output as well as attention mask.
- The attention mask is compared with pseudo attention mask extracted from pre-trained CAM.
- Final output is synthesized by alpha-blending coarse result and input image using the predicted attention map. (*VitON-like*)
- Reference image is chosen by attribute similarity, and perceptual loss is computed.

### Discriminator side

- Two discriminators: image discriminator and crop discriminator.
- Image D takes in full images and inspects real / fake.
- Crop discriminator takes patch extracted from attention map (attribute-related region) and evaluates the local region.
- Wasserstein regularization term is added to stabilize training.
- In addition to real / fake decision, classification is also done using separate head (*I guess..?*)

### Objective

- Discriminator 
<img width="291" alt="image" src="https://user-images.githubusercontent.com/46308407/162160510-700741f8-fa5a-4a10-bb5d-9002431ff945.png">

- Generator: adversarial + classification + cycle consistency (*so that unrelated regions remain unchanged*) + attention mask + perceptual
<img width="618" alt="image" src="https://user-images.githubusercontent.com/46308407/162160575-167da752-b370-47fe-a74a-0d0402cc5ff5.png">

### Elaboration on  region-specific editing

<img width="703" alt="image" src="https://user-images.githubusercontent.com/46308407/162161002-74beaa3d-46e6-476a-bb44-a0a9a24fa82e.png">

## Evaluation

- CycleGAN-like architecture
- DeepFashion and Shopping100K dataset

<img width="716" alt="image" src="https://user-images.githubusercontent.com/46308407/162161411-b0510cc4-7e39-4c2d-a827-90c89e8a3b2e.png">
<img width="713" alt="image" src="https://user-images.githubusercontent.com/46308407/162161549-9d63a7b3-5a60-4aee-bc3d-dd556039c61e.png">
<img width="1454" alt="image" src="https://user-images.githubusercontent.com/46308407/162161691-13033f06-65aa-46ca-9c29-435695a75f96.png">
<img width="697" alt="image" src="https://user-images.githubusercontent.com/46308407/162161757-afda086d-62c4-4122-a17c-048b77f439d4.png">
![Uploading image.png…]()

## Comments

- Looks like the authors have cleverly borrowed insights from Virtual Try-on literature (like alpha-blending)
- Providing additional signal for attention map learning using CAM is quite interesting, though I have doubts about the accuracy of CAM.
- The use of cycle consistency loss to prevent changes unrelated regions is also interesting, and I wonder if it helped. It will slow down training for sure.
